### ISSUE TITLE
Ability to replace themed blog colors for reverted messaging text bubble contents with a consistent set of colors

### ISSUE THREAD

**Issue by citrusella** [enchanted-sword/dashboard-unfucker#162]
**Describe the feature**

Was trying to write my own CSS file to make some slight changes to the CSS of the new messaging window on top of what this style provides (because they're minor quirks I just don't like, for instance the height of the window header and the white circle around the icons in the bottom right, and not something I expected this script to take care of).

Though this piece of information is otherwise not important to the thing I'm suggesting, I *was* working off a tab I still have open with the old messaging window in it, and I noticed my conversations in the old style all had white message bubbles with black text and went "oh, I'd like to implement that in the CSS style because I like the consistency"... except that after several tries, I couldn't, and I'm *fairly* sure it's due to something this script has implemented regarding message bubble color being considered more important in the hierarchy CSS goes through? (I am mostly inferring this is from the script and not Tumblr itself due to the colors when the message reverting switch is disabled being totally different, alongside the human-readable ID on the selector. XP) It's possible this inability to be styled using a user CSS style might better be described as a bug, but I wasn't sure and decided to lean toward more of a feature-request type of thinking.

After attempting to force it to work several times (and some decidedly undesirable attempts at workarounds) I figured it was worth asking here (rather than, say, jumping to doing something like manually modifying my copy of the script): I think it would be nice to have the option to be able to use a consistent set of colors (whether black-on-white, or customizable, or something else?) as opposed to the colors being different from conversation to conversation due to blog theming.

Thanks for taking a look at my idea!

**Additional context**
A message bubble in the old layout, whose colors I was trying to emulate (as far as I can tell, all bubbles in the old layout are black-on-white):

![A message bubble reading "test" with black text on a white background. No display name is visible, but an avatar can be seen to the left of the message.](https://github.com/enchanted-sword/dashboard-unfucker/assets/14182450/92a4e37b-522e-422f-94e5-b05e2e030120)

The same message bubble in the new layout, which appears to inherit colors related to the blog it's replying to, and which does not seem to allow the colors to be styled using a user CSS style (i.e. through Stylus):

![A message bubble reading "test" with light text on a dark background. A display name is visible, and no avatar can be seen to the left of the message.](https://github.com/enchanted-sword/dashboard-unfucker/assets/14182450/ee4851a3-f6c8-4c02-9537-02b2aa46d34d)

**Comment by enchanted-sword** [comment]
That's a good idea to bring up!
You may not have encountered this, since I find it's the custom colour people change the least out of the three, but in the old messaging layout, the text in the message bubbles was coloured based on the title colour of the blog you were messaging. The default title colour is of course, regular old #000000, and so most blogs would display black on white. You can see the differences here in this old screenshot of a conversation between a mutual (title colour #000000) and I (title colour #c97dff):
![image](https://github.com/enchanted-sword/dashboard-unfucker/assets/81981111/3d03edd0-a7e9-470e-b470-e967cc4a1302)
![image](https://github.com/enchanted-sword/dashboard-unfucker/assets/81981111/1744cd6d-8620-4c21-a4b6-83d83f84eb9d)
The current implementation in the script is as follows:
```js
const styleMessaging = conversations => {
        for (const conversation of conversations) {
          fetchOtherBlog(conversation).then(({headerImageFocused, backgroundColor, titleColor, linkColor, otherParticipantName, selectedBlogName}) => {
            conversation.id = `messaging-${otherParticipantName}`
            conversation.style.backgroundColor = backgroundColor;
            const style = document.createElement("style");
            style.classList.add("customMessagingStyle");
            conversation.append(style);
            const msgBackground = contrastBW(titleColor);
            const tsColor = contrastBW(backgroundColor);
            if (contrast(linkColor, backgroundColor) > 0.33) {
              linkColor = tsColor;
            }
            titleColor = rgbToString(hexToRgb(titleColor));
            style.innerText = `
              #messaging-${otherParticipantName} ${keyToCss("headerWrapper")} { background: no-repeat top/100% url(${headerImageFocused}) }
              #messaging-${otherParticipantName} ${keyToCss("messageText")}${keyToCss("ownMessage")} ${keyToCss("messageHeader")}::before { content: "${selectedBlogName}"; }
              #messaging-${otherParticipantName} ${keyToCss("messageText")}:not(${keyToCss("ownMessage")}) ${keyToCss("messageHeader")}::before { content: "${otherParticipantName}"; }
              #messaging-${otherParticipantName} ${keyToCss("statusWithCaption")} { color: ${linkColor} !important; }
              #messaging-${otherParticipantName} ${keyToCss("timestamp")} { color: rgba(${tsColor},.6) }
              #messaging-${otherParticipantName} ${keyToCss("name")}, #messaging-${otherParticipantName} ${keyToCss("description")}, 
              #messaging-${otherParticipantName} ${keyToCss("descriptionContainer")} { color: rgb(${tsColor}) !important; }
              #messaging-${otherParticipantName} ${keyToCss("messageText")}, #messaging-${otherParticipantName} ${keyToCss("messagePost")} ${keyToCss("header")},
              #messaging-${otherParticipantName} ${keyToCss("headerPreview")} ${keyToCss("action")} {
                background: rgb(${msgBackground}) !important;
                color: rgb(${titleColor}) !important;
              }
              #messaging-${otherParticipantName} ${keyToCss("headerPreview")} ${keyToCss("summary")} { color: rgba(${titleColor}, .6) !important; }
            `;
          });
        }
      };
```
The custom id is used so that the stylesheet doesn't affect anything outside the messaging window, and the rules do make use of `!important` to take precedence over the default rules, hence why you would likely be unable to style them using something like Stylus.

All that being said, while using the title colour for the message text is what the old messaging layout did, and the script does automatically optimize for colour contrast, it would definitely be nice to have the ability to set a custom colour, so I will implement that in the future.

**Comment by citrusella** [comment]
Yep, that explanation lines up with what I'd thought was going on, glad I wasn't off-base!

...Also... huh! I am *very* surprised I never encountered that even once over the years and am now confused why I've *never* seen a color change even though you've said it's rare, especially since as far as I can tell the blog I used in my screenshots (one of my own sideblogs) *does* have a changed title color. 🤔 ~~Unless I'm thinking of something completely different. I'm good at doing that. XD~~ But good to know, and thanks for the response

**Comment by enchanted-sword** [comment]
Added in v5.3.3
